src/mds/MDSRank.cc

```diff
@@ -2903,14 +2909,18 @@ void MDSRankDispatcher::handle_asok_command(
  * MDSRank after calling it (we could have gone into shutdown): just
  * send your result back to the calling client and finish.
  */
-void MDSRankDispatcher::evict_clients(
+int MDSRankDispatcher::evict_clients(
   const SessionFilter &filter,
-  std::function<void(int,const std::string&,bufferlist&)> on_finish)
+  std::function<void(int,const std::string&,bufferlist&)> on_finish, bool yes_i_really_mean_it)
 {
+  if (!yes_i_really_mean_it) {
```

Comment:
What is the point of passing the flag to this method if you are just bailing immediately?  I thought @vshankar's proposal was to require `--yes-i-really-mean-it` only for a blank `filter` (i.e. when it matches everything)?
> [...] add `--yes-i-really-mean-it` to `client evict` command without which clients are not evicted (when no specific client-id is provided).

Reply:
Right, the flag is certainly meant to be used when no client-id filter is provided.